Formula/go.rb

```diff
@@ -1,15 +1,16 @@
 class Go < Formula
   desc "Open source programming language to build simple/reliable/efficient software"
   homepage "https://go.dev/"
-  url "https://go.dev/dl/go1.20.6.src.tar.gz"
-  mirror "https://fossies.org/linux/misc/go1.20.6.src.tar.gz"
-  sha256 "62ee5bc6fb55b8bae8f705e0cb8df86d6453626b4ecf93279e2867092e0b7f70"
+  url "https://go.dev/dl/go1.21rc3.src.tar.gz"
+  mirror "https://fossies.org/linux/misc/go1.21rc3.src.tar.gz"
+  sha256 "9de96a2f15196a9c69cb9ab122c9c2aaecca30ec9448caa71543eaeb19ca91ab"
```

Comment:
To be updated with hash from https://go.dev/dl/ once 1.21.0 is released. Check if filenames also have the ".0" suffix!
```suggestion
  url "https://go.dev/dl/go1.21.0.src.tar.gz"
  mirror "https://fossies.org/linux/misc/go1.21.0.src.tar.gz"
  sha256 "...TODO..."
```